### PR TITLE
Make verbosity configurable and not leak sensitive values

### LINKED ISSUE #4247: RClone logs sensitive values by default
## Describe the bug
An earlier PR (https://github.com/SeldonIO/seldon-core/pull/3203) increased the verbosity that rclone logs with, which, in turns happens to also log sensitive values like S3 secret key configured via environment variables:

```
2022/08/01 17:27:54 DEBUG : Setting access_key_id="<sensitive value>" for "s3" from environment variable RCLONE_CONFIG_S3_ACCESS_KEY_ID
2022/08/01 17:27:54 DEBUG : Setting secret_access_key="<sensitive value>" for "s3" from environment variable RCLONE_CONFIG_S3_SECRET_ACCESS_KEY
```

Would it be possible to make the verbosity configurable and verbosity level 1 (`-v`) by default?  The verbosity can also be configured with `RCLONE_VERBOSE=(0|1|2)`, but, since the argument `-vv` is explicitly passed in the Dockerfile, this gets ignored.  I currently can't figure out a workaround for this while I wait for a fix.

## To reproduce
1. Configure Seldon Core to use AWS S3 secrets from a Secret resource that specifies `RCLONE_CONFIG_S3_...` config values
2. Deploy a model that references this AWS S3 secret
3. Check the logs for the initializer container (in my case it was a tensorflow model, so the container was `tfserving-model-initializer`)

## Expected behaviour
By default, logs don't log DEBUG messages, which may expose sensitive values, but, verbosity is configurable if needed for debugging

## Environment
* Cloud Provider: GKE
* Kubernetes Cluster Version [Output of `kubectl version`] 
```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersio
n:"go1.16.1", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.8-gke.202", GitCommit:"88deae00580af268497b9656f216cb092b630563", GitTreeState:"clean", BuildDate:"2022-06-03T03:27:52Z",
GoVersion:"go1.16.14b7", Compiler:"gc", Platform:"linux/amd64"}
```
* Deployed Seldon System Images: [Output of `kubectl get --namespace seldon-system deploy seldon-controller-manager -o yaml  | grep seldonio`]
```
docker.io/seldonio/seldon-core-operator:1.14.0
```

## Model Details
Tensorflow model from a private repo

### PR DESCRIPTION
**Pull request by gtrog**
**What this PR does / why we need it**:
Reduce verbose logging, which leaks sensitive values like secrets/keys, but, make it configurable via `RCLONE_VERBOSE`.

**Which issue(s) this PR fixes**:
Fixes #4247

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @gtrog. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **majolo**
You can assign the PR to them by writing `/assign @majolo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtrog** [comment]
Tested locally:

```
# docker build . -t rclone
```

Default verbosity, no `DEBUG` messages:
```
# docker run -it --rm rclone
Usage:
  rclone copy source:path dest:path [flags]

Flags:
      --create-empty-src-dirs   Create empty source dirs on destination after copy
  -h, --help                    help for copy

Use "rclone [command] --help" for more information about a command.
Use "rclone help flags" for to see the global flags.
Use "rclone help backends" for a list of supported services.
Command copy needs 2 arguments minimum: you provided 0 non flag arguments: []
```

Then, overriden verbosity contains `DEBUG` messages:
```
# docker run -it --rm -e RCLONE_VERBOSE=2 rclone
2022/08/01 18:40:27 DEBUG : rclone: Version "v1.57.0" starting with parameters ["rclone" "copy"]
Usage:
  rclone copy source:path dest:path [flags]

Flags:
      --create-empty-src-dirs   Create empty source dirs on destination after copy
  -h, --help                    help for copy

Use "rclone [command] --help" for more information about a command.
Use "rclone help flags" for to see the global flags.
Use "rclone help backends" for a list of supported services.
Command copy needs 2 arguments minimum: you provided 0 non flag arguments: []
```

**Comment by RafalSkolasinski** [comment]
Thank you @gtrog, the change looks good. Could you post difference in output between verbose level 1 and 2 for completness?

**Comment by RafalSkolasinski** [comment]
Thanks @gtrog, I'd probably prefer to have some verbosity 2.5 (knowing what vars are set but with values redacted automatically). But as it seems there's no option for this setting for verbosity 1 seems appropriate.

**Review by RafalSkolasinski — APPROVED** [review]
LGTM

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[RafalSkolasinski](https://github.com/SeldonIO/seldon-core/pull/4249#pullrequestreview-1063125139 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [RafalSkolasinski]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtrog** [comment]
> Thanks @gtrog, I'd probably prefer to have some verbosity 2.5 (knowing what vars are set but with values redacted automatically). But as it seems there's no option for this setting for verbosity 1 seems appropriate.

Yeah, I'll see if I can make a contribution to RClone as well and maybe eventually circle back here and do that

**Pull request merged**
